src/frontend/src/common/redux/types.ts

```diff
@@ -1,11 +1,13 @@
 import { ExposedNotificationProps } from "czifui";
 import { NotificationComponents } from "src/components/NotificationManager/components/Notification";
 
-export type ActionType<T> = (payload?: T) => {
-  type: CZGEReduxActions;
-  payload?: T;
+export type ComplexActionType<T, U> = (payload: T) => {
```

Comment:
So `ComplexActionType` is for one that mutates the incoming payload from "raw" input payload `T` to "cooked" output payload `U`?

Reply:
yep